lib/Backend/BailOut.cpp

```diff
@@ -2430,7 +2430,16 @@ void BailOutRecord::ScheduleLoopBodyCodeGen(Js::ScriptFunction * function, Js::S
                 break;
 
             case IR::BailOutOnNotNativeArray:
-                rejitReason = RejitReason::ExpectingNativeArray;
+                // REVIEW: We have an issue with array profile info.  The info on the type of array we have won't
+                //         get fixed by rejitting.  For now, just give up after 50 rejits.
```

Comment:
Will you leave this comment in (or convert to a TODO)? Will you open a longer-term work item to look into this?